### ISSUE TITLE
False warning of function never used

### ISSUE THREAD

**Issue by catenacyber** [rust-lang/rust#110588]
I tried this code:

https://github.com/OISF/suricata/pull/8744/commits/1198fa40a0d77a3717ae5c06d36f007cfccf8ef8

I expected to see this happen: `cargo clippy --all-features` does not report any problem

Instead, this happened:
`cargo clippy --all-features`

reports
```
    Checking suricata v7.0.0-rc2-dev (/Users/catena/catena/prod/suricata/rust)
error: associated function `new` is never used
  --> src/telnet/telnet.rs:45:8
   |
45 |     fn new() -> Self {
   |        ^^^
   |
note: the lint level is defined here
  --> src/lib.rs:18:38
   |
18 | #![cfg_attr(feature = "strict", deny(warnings))]
   |                                      ^^^^^^^^
   = note: `#[deny(dead_code)]` implied by `#[deny(warnings)]`

error: could not compile `suricata` due to previous error
```

If I comment out this function, I get 
```
error[E0599]: no function or associated item named `new` found for struct `TelnetTransaction` in the current scope
   --> src/telnet/telnet.rs:138:41
    |
39  | pub struct TelnetTransaction {
    | ---------------------------- function or associated item `new` not found for this struct
...
138 |         let mut tx = TelnetTransaction::new();
    |                                         ^^^ function or associated item not found in `TelnetTransaction`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `new`, perhaps you need to implement one of them:
            candidate #1: `lexical_core::util::sequence::VecLike`
            candidate #2: `digest::core_api::VariableOutputCore`
            candidate #3: `digest::Digest`
            candidate #4: `digest::Mac`
            candidate #5: `digest::VariableOutput`
            candidate #6: `digest::KeyInit`
            candidate #7: `digest::crypto_common::KeyIvInit`
            candidate #8: `digest::typenum::Bit`
            candidate #9: `brotli::enc::hash_to_binary_tree::Allocable`
            candidate #10: `brotli::enc::threading::AnyBoxConstructor`
            candidate #11: `aes::NewBlockCipher`
            candidate #12: `aes::cipher::NewCipher`
            candidate #13: `aes_gcm::NewAead`
            candidate #14: `universal_hash::NewUniversalHash`
```

It is used indeed

### Meta

`rustc --version --verbose`:
```
rustc 1.68.0-nightly (0442fbabe 2023-01-10)
binary: rustc
commit-hash: 0442fbabe24ec43636a80ad1f40a0ad92a2e38df
commit-date: 2023-01-10
host: x86_64-apple-darwin
release: 1.68.0-nightly
LLVM version: 15.0.6
```
